infra/src/main/kotlin/fi/fta/geoviite/infra/map/AlignmentGeometry.kt

```diff
@@ -140,26 +140,36 @@ fun simplify(
     val isOverResolution = { mValue: Double -> resolution?.let { r -> (mValue - previousM).roundToInt() >= r } ?: true }
     return segments
         .flatMapIndexed { sIndex, s ->
-            val isEndPoint = { pIndex: Int ->
-                val isTrackEndPoint =
-                    (sIndex == 0 && pIndex == 0) ||
-                        (sIndex == segments.lastIndex && pIndex == s.segmentPoints.lastIndex)
-                val isSegmentStartPoint = pIndex == 0
-                isTrackEndPoint || includeSegmentEndPoints && isSegmentStartPoint
-            }
-            val isSegmentEndPoint = { pIndex: Int -> pIndex == 0 || pIndex == s.segmentPoints.lastIndex }
-            val bboxContains = { pIndex: Int ->
-                bbox == null || s.segmentPoints.getOrNull(pIndex)?.let(bbox::contains) ?: false
-            }
-            s.segmentPoints.mapIndexedNotNull { pIndex, p ->
-                if (isPointIncluded(pIndex, p.m + s.startM, isEndPoint, isOverResolution, bboxContains)) {
-                    if (!isSegmentEndPoint(pIndex)) {
-                        // segment end points should be additional points,
-                        // so increase m-counter only when handling middle points
-                        previousM = s.startM + p.m
-                    }
-                    p.toAlignmentPoint(s.startM)
-                } else null
+            if (sIndex == 0 || sIndex == segments.lastIndex || isOverResolution(s.endM)) {
```

Comment:
Jos raide koostuu pienistä segmenteistä ja endpointteja ei ole pyydetty, tämä päätyy suodattamaan koko geometrian pois. Tuossa isOverResolution pitäisi summata mukaan aiemmat segmentit jotka on skipattu samasta syystä. En tiedä kuinka usein tuo käytännössä ilmenee, mutta tuntuu että tämän logiikan voisi varmaan muotoilla myös jotenkin niin että raiteen pituutta hypitään resoluution verran eteenpäin ja katsotaan mihin segmenttiin se osuu, tms. :thinking: